shared_model/backend/protobuf/queries/proto_query.hpp

```diff
@@ -139,10 +109,33 @@ namespace shared_model {
       }
 
      private:
+      /// lazy variant shortcut
+      template <typename T>
+      using Lazy = detail::LazyInitializer<T>;
+
+      using LazyVariantType = Lazy<ProtoQueryVariantType>;
       // ------------------------------| fields |-------------------------------
       // lazy
-      const LazyVariantType variant_{
-          [this] { return loadQuery<ProtoQueryListType>(*proto_); }};
+      const LazyVariantType variant_{[this] {
+        auto &&ar = *proto_;
+        if (not ar.has_payload()) {
+          throw std::invalid_argument("Query missing payload");
```

Comment:
Is it necessary to use exceptions?

Reply:
It is possible to use assertions here, since these exceptions are for debugging purposes.

Reply:
We have to handle it somehow

Reply:
Actually not only for debug. Check ```ModelQueryBuilderTest``` - it handles case for possible missing payload. It can be triggered from client libraries so exception is only possible solution here.

Reply:
@luckychess Yes, you are correct, my mistake. Since I moved these checks to builder, it should still be OK.